beancount_reds_importers/libtransactionbuilder/investments.py

```diff
@@ -91,6 +91,14 @@ def build_account_map(self):
             "dep":          self.config['transfer'],
         }
 
+    def build_metadata(self, file, counter, metadata={}):
```

Comment:
My understanding is that `ot` is fairly specific to 'transaction', and other named vars may be needed for other metatypes

Although the ordering technically doesn't matter much for named args, perhaps it'd make sense to have `metaType` before `ot` such that one would have `def build_metadata(self, file, counter, metadata={}, metatype=None, ot=None)` and the upcoming PR for accounts/commodities/etc. would add on to this as appropriate and become
`def build_metadata(self, file, counter, metadata={}, metatype=None, ot=None, commodity=None, account=None)` - **would that work?**

For reference, here was the original way I was envisioning this in fidelity_csv_old was with separate functions: https://github.com/savingsandloan/beancount_reds_importers/blob/7ce5223681b2c539dc0b8418b81bd0fc33ce46cb/beancount_reds_importers/libtransactionbuilder/investments.py#L99

Reply:
How about simply:
```
def build_metadata(self, file, transaction, metatype=None, data={})
```

- additional vars like `commodity` and `account` can be passed in via `data`.
- `metadata={}` will not be passed in: the caller will instead do `meta  |= self.build_metadata(...` if needed

Reply:
Just double checking, are you inferring from `def build_metadata(self, file, transaction, metatype=None, data={})` that:
- `counter` data wouldn't be passed in? Guessing that's just a typo.
- `transaction` data would always be required? I'm thinking that should either be part of `data` (assuming one would already be setting metatype to `transaction`), or default as `transaction=None` for when a build_metadata call doesn't have any transaction info and is relying more on metatype+data.

Reply:
> counter data wouldn't be passed in?

Correct, it would not. `counter` can be handled by the caller.

> transaction data would always be required?

Agree, better to stick `transaction` into `data` instead.